src/scripts/main.js

```diff
@@ -2,10 +2,22 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function correctly returns a promise that resolves when the specified event occurs on the element. However, ensure that the `element` and `eventName` passed to this function are valid and exist in the DOM to avoid runtime errors.